dcc-portal-ui/app/scripts/genesets/js/genesets.js

```diff
@@ -183,145 +182,29 @@
-                  var name = drug.name;
-                  var genes = drug.genes;
-                  _.forEach(genes, function(gene) {
-                    var uniprotId = gene.uniprot;
-                    
-                    if (_.isUndefined(drugMap[uniprotId])) {
-                      drugMap[uniprotId] = [];
-                    }
-                    
-                    drugMap[uniprotId].push({
-                      'zincId': zincId,
-                      'name': name
-                    });
-                  });
-                });
-
-                var pathwayDrugHighlights = [];
-                // Normalize into array
-                _.forEach(map,function(value,id) {
-                  if(value && value.dbIds && drugMap[id]) {
-                    pathwayDrugHighlights.push({
-                      uniprotId:id,
-                      dbIds:value.dbIds.split(','),
-                      drugs:drugMap[id]
-                    });
-                  }
-                });
-
-                // Get ensembl ids for all the genes so we can link to advSearch page
-                uniprotIds = _.pluck(pathwayDrugHighlights, 'uniprotId');
-                GeneSetVerificationService.verify( uniprotIds.join(',') ).then(function(data) {
-                  _.forEach(pathwayDrugHighlights,function(n){
-
-                    var geneKey = 'external_db_ids.uniprotkb_swissprot';
-                    if (! data.validGenes[geneKey]) {
-                      return;
-                    }
-
-                    var uniprotObj = data.validGenes[geneKey][n.uniprotId];
-                    if(!uniprotObj){
-                      return;
-                    }
-                    n.geneSymbol = uniprotObj[0].symbol;
-                    n.geneId = uniprotObj[0].id;
-                  });
-                });
-                _ctrl.pathway.mutationHighlights = pathwayMutationHighlights;
-                _ctrl.pathway.drugHighlights = pathwayDrugHighlights;
-              });
-            });
-            
-            // Wait before rendering legend, 
-            // Same approach taken in the pathway viewer page. 
-            setTimeout(function () {
-              $scope.$broadcast(PathwaysConstants.EVENTS.MODEL_READY_EVENT, {});
-            }, 100);
+          _ctrl.pathway = {}; // initialize pathway object
+
+          PathwayDataService.getPathwayData(geneSet.id, null)
+            .then(function (pathwayData) {
+              _ctrl.pathway = {
+                xml: pathwayData.xml,
+                zooms: pathwayData.zooms,
+                mutationHighlights: pathwayData.mutationHighlights,
+                drugHighlights: pathwayData.drugHighlights
+              };
+
+              // Wait before rendering legend, 
+              // Same approach taken in the pathway viewer page. 
+              setTimeout(function () {
+                $scope.$broadcast(PathwaysConstants.EVENTS.MODEL_READY_EVENT, {});
+              }, 100);
```

Comment:
Grab me next time you're on this side of the building and let's see if we can get rid of both these timeouts